content_server/openttd/tcp_content.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+import logging
+
+from .protocol.exceptions import PacketInvalid
+from .protocol.source import Source
+from .receive import OpenTTDProtocolReceive
+from .send import OpenTTDProtocolSend
+
+log = logging.getLogger(__name__)
+
+
+class OpenTTDProtocolTCPContent(asyncio.Protocol, OpenTTDProtocolReceive, OpenTTDProtocolSend):
+    def __init__(self, callback_class):
+        super().__init__()
+        self._callback = callback_class
+        self._callback.protocol = self
+        self._queue = asyncio.Queue()
+        self._data = b""
+        self.is_ipv6 = None
+
+        asyncio.create_task(self._process_queue())
+
+    def connection_made(self, transport):
+        self.transport = transport
+        socket_addr = transport.get_extra_info("peername")
+        self.source = Source(self, socket_addr, socket_addr[0], socket_addr[1])
+        # TODO -- Support PROXY protocol
+
+        if len(transport.get_extra_info("sockname")) == 4:
+            self.is_ipv6 = True
+        else:
+            self.is_ipv6 = False
```

Comment:
```suggestion
        self.is_ipv6 = (len(transport.get_extra_info("sockname")) == 4)
```
Though... `== 4` ? Not 6? Is that correct?

Reply:
Haha, misleading, isn't it? A socket in Python that is IPv4 is a tuple of two. A socket in Python that is IPv6 is a tuple of four. I will supply it with a nice comment :D